test/e2e/util/eventually.go

```diff
@@ -0,0 +1,396 @@
+package util
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	certificatesv1alpha1 "github.com/openshift/hypershift/api/certificates/v1alpha1"
+	hyperv1 "github.com/openshift/hypershift/api/hypershift/v1beta1"
+	certificatesv1 "k8s.io/api/certificates/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func defaultOptions() *EventuallyOptions {
+	return &EventuallyOptions{
+		interval:       100 * time.Millisecond,
```

Comment:
this seems very short for e2e.  there are 6000 100ms intervals in a 10m timeout period.  why not 1s? or 5s?  we don't want to DoS the KAS.

Reply:
KAS can handle many more orders of magnitude, so I am not worried about it - and the longer the poll, the longer every single test must take as a lower bound.